RealmWeaver/ModuleWeaver.cs

```diff
@@ -163,12 +163,24 @@ public void Execute()
         _wovenPropertyAttributeConstructor = ModuleDefinition.ImportReference(wovenPropertyAttributeClass.GetConstructors().First());
 
         _corLib = ModuleDefinition.AssemblyResolver.Resolve((AssemblyNameReference)ModuleDefinition.TypeSystem.CoreLibrary);
-        System_Object = _corLib.MainModule.GetType("System.Object");
-        System_Boolean = _corLib.MainModule.GetType("System.Boolean");
-        System_String = ModuleDefinition.ImportReference(_corLib.MainModule.GetType("System.String"));
-        System_Type = ModuleDefinition.ImportReference(_corLib.MainModule.GetType("System.Type"));
-        // WARNING the GetType("System.Collections.Generic.List`1") below RETURNS NULL WHEN COMPILING A PCL
-        System_IList = ModuleDefinition.ImportReference(_corLib.MainModule.GetType("System.Collections.Generic.List`1"));
+        System_Object = ModuleDefinition.TypeSystem.Object; 
+        System_Boolean = ModuleDefinition.TypeSystem.Boolean;
+        System_String = ModuleDefinition.TypeSystem.String;
+        var typeTypeDefinition = _corLib.MainModule.GetType("System.Type");
+        if (typeTypeDefinition == null)
+        {
+            typeTypeDefinition = _corLib.MainModule.ExportedTypes.First(t => t.FullName == "System.Type").Resolve();
+        }
+        System_Type = ModuleDefinition.ImportReference(typeTypeDefinition);
+
+        var listTypeDefinition = _corLib.MainModule.GetType("System.Collections.Generic.List`1");
+        if (listTypeDefinition == null)
```

Comment:
Comment here please emphasising this is primarily for PCL - otherwise someone may wonder why if they never see this branch needed.